src/components/TransactionBlock/PlrDaoStakingTransactionBlock.tsx

```diff
@@ -369,15 +379,16 @@ const PlrDaoStakingTransactionBlock = ({
   );
 
   const getNftList = async () => {
+    if (!accountAddress || !providerAddress || !sdk) return;
     try {
-      if (!accountAddress || !providerAddress || !sdk) return;
-      const output = await sdk.getNftList({
-        account: accountAddress || providerAddress,
-      });
-      let hasNFTContractAddress = output?.items?.filter((nft) => nft.contractAddress === plrDaoMemberNFT.address);
-      if (hasNFTContractAddress?.length) {
-        setIsNFTMember(true);
-      }
+      const [providerNFTDetails, accountNFTDetails] = await Promise.all([
+        getNftsForChainId(CHAIN_ID.POLYGON, providerAddress, true),
+        getNftsForChainId(CHAIN_ID.POLYGON, accountAddress, true),
+      ]);
+      const hasMembershipNFT = providerNFTDetails
+        .concat(accountNFTDetails)
+        .some((nft) => addressesEqual(nft.contractAddress, plrDaoMemberNFT.address));
+      setIsNFTMember(hasMembershipNFT);
```

Comment:
I think we can convert this function to a normal arrow function instead of an async functionand do `Promise.all(...).then(res => ...)`

Reply:
@MayukhSamadder2000 imo everything here unless I don't understand your comment
what maybe can be improved here is loop  `[providerAddress, accountAddress]` for `getNftsForChainId`

Reply:
@poocart my point here was why do we need a async function as we're not returning a promise here we can simply use a normal function and maybe just set the values on the then block.
https://betterprogramming.pub/should-i-use-promises-or-async-await-126ab5c98789#:~:text=To%20run%20promises,to%20neglect%20this.